### PR TITLE
Update namespace. Add table_name config to all models

### PR DESCRIPTION
**Pull request by chrispenny**
Prefixed each with `Link_` to make it easier to find all relevant tables, but also because `FileLink` is already used by framework.

### CODE REVIEW

File: src/Models/EmailLink.php [hunk 1]
```diff
@@ -12,6 +12,8 @@
 class EmailLink extends Link
 {
 
+    private static $table_name = 'Link_EmailLink';
```
Comment (mfendeksilverstripe) [inline src/Models/EmailLink.php:15]:
```suggestion
    private static $table_name = 'LinkFieldLink_EmailLink';
```

File: src/Models/ExternalLink.php [hunk 2]
```diff
@@ -10,11 +10,12 @@
 class ExternalLink extends Link
 {
 
+    private static $table_name = 'Link_ExternalLink';
```
Comment (mfendeksilverstripe) [inline src/Models/ExternalLink.php:13]:
```suggestion
    private static $table_name = 'LinkFieldLink_ExternalLink';
```

File: src/Models/FileLink.php [hunk 3]
```diff
@@ -8,17 +8,19 @@
...
 /**
  * A link to a File track in asset-admin
+ *
  * @property File $File
  * @property int $FileID
  */
 class FileLink extends Link
 {
 
+    private static $table_name = 'Link_FileLink';
```
Comment (mfendeksilverstripe) [inline src/Models/FileLink.php:18]:
```suggestion
    private static $table_name = 'LinkFieldLink_FileLink';
```

File: src/Models/Link.php [hunk 4]
```diff
@@ -21,6 +21,8 @@
 class Link extends DataObject implements JsonData, Type
 {
 
+    private static $table_name = 'Link';
```
Comment (mfendeksilverstripe) [inline src/Models/Link.php:24]:
```suggestion
    private static $table_name = 'LinkFieldLink';
```

File: src/Models/SiteTreeLink.php [hunk 5]
```diff
@@ -16,6 +16,8 @@
 class SiteTreeLink extends Link
 {
 
+    private static $table_name = 'Link_SiteTreeLink';
```
Comment (mfendeksilverstripe) [inline src/Models/SiteTreeLink.php:19]:
```suggestion
    private static $table_name = 'LinkFieldLink_SiteTreeLink';
```

**Review by mfendeksilverstripe — CHANGES_REQUESTED** [review]
I think we should go with a less generic table names to prevent potential naming conflicts. I can definitely see table called `Link` having a conflict especially if a project is migrating from another module. I added `LinkFieldLink` prefix as the name of this module is `LinkField`.

**Review by mfendeksilverstripe — CHANGES_REQUESTED** [review]
@chrispenny The changes look good but it seems some of the functionality is broken because of this change. This is my test setup (TNZ core after updating namespace references):

* Block page with a TextBlock / ButtonLink
* I switcher the block to NOT be inline editable
* navigated to the block edit form

Expected:

Screen Shot 2021-11-16 at 8 45 37 AM

Actual:

Screen Shot 2021-11-16 at 8 42 20 AM

Maybe there are some class references in the React bundle as well?

**Comment by mfendeksilverstripe** [comment]
@chrispenny Can you please check if you are able to save and retain link related changes? I can see that the UI now works but none of the changes I make are retained. Seems to be an issue on both inline edit and normal edit form cases.

Create a link

Screen Shot 2021-11-16 at 11 41 11 AM

Hit save

Screen Shot 2021-11-16 at 11 41 20 AM

Link doesn't show up even after page reload.

**Comment by chrispenny** [comment]
Hi @mfendeksilverstripe,

These seem to be working ok for me.

Inline editor (text block):
Screen Shot 2021-11-16 at 1 29 23 PM

Standard editor (cta block), straight after a publish:
Screen Shot 2021-11-16 at 1 30 49 PM

Standard editor (cta block), after a refresh:
Screen Shot 2021-11-16 at 1 31 01 PM

I don't believe I have done anything special other than a `dev/build` and appropriate flushes.

**Review by mfendeksilverstripe — APPROVED** [review]
Tested on local and it's working fine 👍 
thanks for contribution :)

**Comment by mfendeksilverstripe** [comment]
@chrispenny Feel free to merge.

**Pull request merged**